scijava/scijava-ops-engine/src/main/java/org/scijava/ops/engine/matcher/impl/DefaultRichOp.java

```diff
@@ -1,46 +1,29 @@
 
 package org.scijava.ops.engine.matcher.impl;
 
-import java.lang.reflect.Type;
-import java.util.UUID;
-
-import org.scijava.ops.api.Hints;
 import org.scijava.ops.api.OpExecutionSummary;
-import org.scijava.ops.api.OpHistory;
-import org.scijava.ops.api.OpInfo;
 import org.scijava.ops.api.OpMetadata;
 import org.scijava.ops.api.RichOp;
 import org.scijava.ops.api.features.BaseOpHints.DependencyMatching;
 
-public class DefaultRichOp implements RichOp {
+public abstract class DefaultRichOp implements RichOp {
```

Comment:
https://github.com/scijava/scijava/issues/81#issuecomment-987189209

Reply:
Doing so might cause a bit of a packaging nightmare.

It is the `OpWrapper`'s job to return a `RichOp` from an `OpInstance`. `OpWrapper` needs to live in `org.scijava.ops.api`, thus we need `RichOp` in SciJava Ops API.

If we made this change, we would need to move `DefaultRichOp` (or `AbstractRichOp`) to SciJava Ops API. This is no good, because it will bring in additional dependencies. (for example, looking at `AbstractOpEnvironment`, we'd impose a dependency on SciJava Progress.)